Prevent Oracle from committing uncommitted transactions on connection close (#45301)
quarkusio/quarkus: docs/src/main/asciidoc/datasource.adoc

In the above example, the type is `AgroalDataSource`, a `javax.sql.DataSource` subtype.
Because of this, you can also use `javax.sql.DataSource` as the injected type.

===== Oracle considerations

As documented in https://github.com/quarkusio/quarkus/issues/36265[issue #36265],

> Oracle has a very weird behavior of committing the uncommitted transactions on connection close.

Reviewer: ```suggestion
Oracle has a very weird behavior of committing the uncommitted transactions on connection closing.
```

Which means that when stopping Quarkus for instance, in progress transactions might be committed even if incomplete.

> Given that is not the expected behavior and that it could lead to data loss, we added an interceptor that rollbacks any unfinished transactions at connection close.

Reviewer: ```suggestion
Given that is not the expected behavior and that it could lead to data loss, we added an interceptor that rolls back any unfinished transactions at connection close.
```

> If this behavior introduced in 3.18 causes issues for your specific workload, you can disable it by setting the `-Dquarkus-oracle-no-automatic-rollback-on-connection-close` system property.

Reviewer: ```suggestion
If this behavior introduced in 3.18 causes issues for your specific workload, you can disable it by setting the `-Dquarkus-oracle-no-automatic-rollback-on-connection-close` system property to `true`.
```
